src/java/org/apache/cassandra/db/guardrails/Guardrails.java

```diff
@@ -129,6 +129,11 @@
                     state -> !CONFIG_PROVIDER.getOrCreate(state).getUserTimestampsEnabled(),
                     "User provided timestamps (USING TIMESTAMP)");
 
+    public static final DisableFlag groupByEnabled =
+    new DisableFlag("group_by",
+                    state -> !CONFIG_PROVIDER.getOrCreate(state).getGroupByEnabled(),
+                    "GROUP BY functionality");
```

Comment:
for me: `GROUP BY functionality is not allowed` will be the error message

Reply:
The `what` argument in the constructor of `DisableFlag` doesn't contain the full message but just the name of what is disallowed, the ` is not allowed` part will be added by the guardrail, [here](https://github.com/apache/cassandra/blob/18bae4628c50ae927da7dfa4e7e1fb8648dd765b/src/java/org/apache/cassandra/db/guardrails/DisableFlag.java#L84). So I think this is correct. Maybe we could add the full message to the calls to `assertFails` in the new test to be completely sure:
```java
assertFails(query, "GROUP BY functionality is not allowed");
```

Reply:
```
@Test
public void checkExplicitlyDisabled() throws Throwable
{
    setGuardrail(false);
    assertFails(query, "GROUP BY functionality is not allowed");
}
```

Modified the unit test to check the full string. Passes.